ipatests/test_webui/test_host.py

```diff
@@ -407,11 +435,353 @@ def test_indirect_associations(self):
+        self.undo_ssh_keys()
+
+        # refresh
+        self.add_sshkey_to_record(host.ssh_rsa, self.data2['pkey'],
+                                  entity=ENTITY, navigate=True, save=False)
+        self.facet_button_click('refresh')
+        self.assert_num_ssh_keys(0)
+
+        # revert
+        self.add_sshkey_to_record(host.ssh_rsa, self.data2['pkey'],
+                                  entity=ENTITY, navigate=True, save=False)
+        self.facet_button_click('revert')
+        self.assert_num_ssh_keys(0)
+
+        # cleanup
+        self.delete(ENTITY, [self.data2])
+
+    @screenshot
+    def test_negative_cert(self):
+        """ Test negative CSR """
+        self.init_app()
+        self.add_record(ENTITY, self.data2)
+        self.close_notifications()
+        self.navigate_to_record(self.data2['pkey'], entity=ENTITY)
+
+        # emtpy CSR
+        csr_add = 'div[name="certificate"] button[name="add"]'
+        csr_add_btn = self.find(csr_add, By.CSS_SELECTOR, strict=True)
+        csr_add_btn.click()
+        self.wait()
+        self.dialog_button_click('ok')
+        self.assert_field_validation_required()
+        self.dialog_button_click('cancel')
+
+        # invalid CSR
+        csr_add = 'div[name="certificate"] button[name="add"]'
+        csr_add_btn = self.find(csr_add, By.CSS_SELECTOR, strict=True)
+        csr_add_btn.click()
+        self.wait()
+        self.fill_textarea('new_cert', host.csr_invalid)
+        self.dialog_button_click('ok')
+        dialog_info = self.get_dialog_info()
+        self.wait()
+        if host.csr_invalid_error in dialog_info['text']:
+            self.dialog_button_click('cancel')
+            self.dialog_button_click('cancel')
+
+        # other hostname CSR
+        self.action_list_action('request_cert', confirm=False)
+        self.assert_dialog()
+        self.fill_text("textarea[name='csr']", host.csr_other_host)
+        self.dialog_button_click('issue')
+        dialog_info = self.get_dialog_info()
+        if host.csr_other_host_error in dialog_info['text']:
+            self.dialog_button_click('cancel')
+            self.dialog_button_click('cancel')
+
+        # cleanup
+        self.delete(ENTITY, [self.data2])
+
+    @screenshot
+    def test_keytab(self):
+        """ Test keytab """
+        self.init_app()
+        self.add_record(ENTITY, self.data2)
+        # provision keytab
+        kt_tmp = '/tmp/test.keytab'
+        hostname = self.data2['pkey']
+        realm = self.config.get('ipa_realm')
+        principal = 'host/{}@{}'.format(hostname, realm)
+        self.run_cmd_on_ui_host('/usr/sbin/ipa-getkeytab '
+                                '-p {} '
+                                '-k {} '.format(principal, kt_tmp))
+        self.navigate_to_record(hostname, entity=ENTITY)
+        self.wait(3)
+        enroll = 'div[name="has_keytab"] label[name="present"]'
+        self.assert_text(enroll, host.krb_enrolled)
+        # test cancel button there bz818665
```

Comment:
It's upstream tests, let's stick to upstream issues. In this case: https://pagure.io/freeipa/issue/1811